test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -282,9 +282,8 @@ func clusterUpgradeWithRuntimeSDKSpec(ctx context.Context, inputGetter func() cl
 func extensionConfig(specName string, namespace *corev1.Namespace) *runtimev1.ExtensionConfig {
 	return &runtimev1.ExtensionConfig{
 		ObjectMeta: metav1.ObjectMeta{
-			// FIXME(sbueringer): use constant name for now as we have to be able to reference it in the ClusterClass.
-			// Random generate later on again when Yuvaraj's PR has split up the cluster lifecycle.
-			//Name: fmt.Sprintf("%s-%s", specName, util.RandomString(6)),
+			// Note: We have to use a constant name here as we have to be able to reference it in the ClusterClass
+			// when configuring external patches.
```

Comment:
Just saw the todo and dropped it. I don't think it's worth the effort to generate the ExtensionConfig name randomly

Reply:
Hmm - this can still be annoying when running manually as I regularly need to manually delete the ExtensionConfig if the test fails unexpectedly.

Could we delete the ExtensionConfig if it exists inside this test? It's mainly a problem because it's a global object.

Reply:
I'm happy to take this as a follow-up